neuro_san/internals/interfaces/environment_configuration.py

```diff
@@ -9,44 +9,38 @@
 # neuro-san SDK Software in commercial settings.
 #
 # END COPYRIGHT
-
 from typing import Any
 from typing import Dict
 
 import os
 
-from neuro_san.internals.run_context.langchain.llms.langchain_llm_client import LangChainLlmClient
-
 
-class LangChainLlmClientFactory:
+class EnvironmentConfiguration:
     """
-    Interface for Factory classes creating LLM client connections for LangChain.
+    Easy policy add on for the get_value_or_env() method for various classes that
+    are effected by configuration via dictionary/hocon and/or environment variables.
     """
 
-    def create_llm_client(self, config: Dict[str, Any]) -> LangChainLlmClient:
-        """
-        Create a LangChainLlmClient instance from the fully-specified llm config.
-        :param config: The fully specified llm config
-        :return: A LangChainLlmClient instance containing run-time resources necessary
-                for model usage by the service.
-
-                Can raise a ValueError if the config's class or model_name value is
-                unknown to this method.
-        """
-        raise NotImplementedError
-
-    def get_value_or_env(self, config: Dict[str, Any], key: str, env_key: str) -> Any:
+    @staticmethod
+    def get_value_or_env(config: Dict[str, Any], key: str, env_key: str,
+                         none_obj: Any = None) -> Any:
```

Comment:
Even though this is a static method, classes that multiply inherit can still use self.<foo> to access this method.  This is more concise for one-line value reading than "EnvironmentConfiguration.get_value_or_env(blah blah)".

At some point we could add a DictionaryExtractor in here to allow for deep keys.